### PR TITLE
Allow separate gitea oauth URL 

### PR DESCRIPTION
**Pull request by qwerty287**
closes https://github.com/woodpecker-ci/woodpecker/issues/3470

### CODE REVIEW

**Comment by xoxys** [comment]
IMO, this should also be named `_DEV_` and outlined that this is not required in most production setup. I would also prefer `OAuthURL` over `OAuth2URL` but no strong feelings about this.

closes https://github.com/woodpecker-ci/woodpecker/issues/3510 seems to be a drive-by fix that might better go in its own PR.

**Comment by xoxys** [comment]
> I don't think so as it's not a dev flag. Maybe we can add some note in docs that most won't need it?

But for what real world scenario this is required? It just applies for dev/test setups, otherwise you have an accessible external domain anyway, or am I wrong?

**Comment by anbraten** [comment]
If I didn't missed anything `WOODPECKER_DEV_OAUTH_HOST` is exactly doing the same already. At least I am using it successfully locally with gitea during development.

**Comment by xoxys** [comment]
Yes me too, and I was wondering as well but had no time to check the details yet.

**Comment by xoxys** [comment]
Yes, the WOODPECKER_GITEA_URL needs to be accessible from the container and from the browser. If you can't find a value that fits both this new option is required. Makes sense to me.

File: server/forge/gitea/gitea.go [hunk 1]
```diff
@@ -63,6 +62,7 @@ type Gitea struct {
 // Opts defines configuration options.
 type Opts struct {
 	URL        string // Gitea server url.
+	OAuth2URL  string // Gitea server url.
```
Comment (xoxys) [inline server/forge/gitea/gitea.go:65]: Better description?

File: server/forge/gitea/gitea_test.go [hunk 2]
```diff
@@ -62,10 +62,6 @@ func Test_gitea(t *testing.T) {
 				g.Assert(f.url).Equal("http://localhost:8080")
 				g.Assert(f.SkipVerify).Equal(true)
 			})
-			g.It("Should handle malformed url", func() {
```
Comment (xoxys) [inline server/forge/gitea/gitea_test.go:65]: Why was this removed?

File: cmd/server/flags.go [hunk 3]
```diff
@@ -386,6 +386,11 @@ var flags = append([]cli.Flag{
 		Name:    "gitea-skip-verify",
 		Usage:   "gitea skip ssl verification",
 	},
+	&cli.StringFlag{
+		EnvVars: []string{"WOODPECKER_DEV_GITEA_OAUTH_URL"},
+		Name:    "gitea-oauth-server",
+		Usage:   "user-facing gitea server address",
```
Comment (xoxys) [inline cmd/server/flags.go:392]:
```suggestion
		Usage:   "user-facing gitea server url for oauth",
```

**Review by xoxys — APPROVED** [review]

File: cmd/server/flags.go (hunk 3 above)
Comment (xoxys) [reply cmd/server/flags.go:392]: Just aligned it to the new description. Feel free to skip this change.
